canonicalwebteam/discourse/parsers/docs_parser.py

```diff
@@ -516,6 +517,19 @@ def _replace_notifications(self, soup):
 
         return soup
 
+    def _add_anchor_support(self, soup):
+        """
+        Adds id anchors to all headers
+        E.g. <h2>Heading</h2> becomes <h2 id="heading">Heading</h2>
+        """
+        for heading in soup.find_all(["h2", "h3", "h4", "h5", "h6"]):
+            heading_id = (
+                heading.text.replace(" ", "-").replace(", ", "-").lower()
```

Comment:
do you think it will be fine with special characters?  (like ?!\"' etc.)

Reply:
Yeah I can add question mark and exclamation, I could cover a lot of characters, I didn't want to make it too complicated and only cover for the docs use case, because we don't use them everywhere, it's just a few situations. 

Sometimes special characters like backticks in the title, just need to be removed because they won't render in the final docs.

Reply:
Testing it, seems that everything works? I also thought special characters don't work
https://codepen.io/carkod/pen/bGwqvJB